### PR TITLE
ASC-373 Fix bug where different decorators would fail

### PR DESCRIPTION
**Pull request by zreichert**
- adds new error classes to except when addressing decorators to test definition
- new test cases that cover examples of differnet kinds of mark decorators

### CODE REVIEW

File: pytest_mark_checker.py [hunk 1]
```diff
@@ -36,14 +36,17 @@ def rule_M501(self, node):
             marked = False
             line_num = node.lineno
             name = node.name
-            if re.search(r'^test_', name):
+            if re.search(r'^test', name):
```
Comment (ryan-rs) [inline pytest_mark_checker.py]: Why did you broaden the regex?

File: pytest_mark_checker.py [hunk 2]
```diff
@@ -36,14 +36,17 @@ def rule_M501(self, node):
...
-                    value = decorator.args[0].s
-                    mark_key = decorator.func.attr
-                    decorator_type = decorator.func.value.value.id
-                    if decorator_type == 'pytest' and mark_key == 'test_id':
-                        marked = True
+                    try:
+                        value = decorator.args[0].s
+                        mark_key = decorator.func.attr
+                        decorator_type = decorator.func.value.value.id
+                        if decorator_type == 'pytest' and mark_key == 'test_id':
```
Comment (ryan-rs) [inline pytest_mark_checker.py]: Should we check the value to make sure it is a valid UUID format?

Comment (zreichert) [reply pytest_mark_checker.py]: I think that is part of the configuration effort.  But yes eventually I think we need to be a able to validate the value provided to a mark.

File: pytest_mark_checker.py (hunk 1 above)
Comment (johnduarte) [reply pytest_mark_checker.py]: Whaat?

File: tests/test_mark_checking.py [hunk 3]
```diff
@@ -30,3 +30,106 @@ def not_a_test():
...
+    (2, 3),
+])
+def test_a_parametrized_mark():
+    pass
+    """)
+    result = flake8dir.run_flake8(extra_args)
+    assert result.out_lines == []
+
+
+def test_do_validate_one_of_many(flake8dir):
```
[the remark this answers is not in the document]
Comment (zreichert) [reply tests/test_mark_checking.py]: The no marks test case already exists.  https://github.com/zreichert/pytest-mark-checker/blob/master/tests/test_mark_checking.py#L17-L23

I'll update to use your suggested case name.

File: tests/test_mark_checking.py [hunk 4]
```diff
@@ -30,3 +30,106 @@ def not_a_test():
...
+@pytest.mark.usefixtures(fixturename1, fixturename2)
+@pytest.mark.test_id('b360c12d-0d47-4cfc-9f9e-5d86c315b1e4')
+def test_a_usefixture_mark():
+    pass
+    """)
+    result = flake8dir.run_flake8(extra_args)
+    assert result.out_lines == []
+
+
+def test_dont_validate_one_of_many(flake8dir):
```
[the remark this answers is not in the document]
Comment (zreichert) [reply tests/test_mark_checking.py]: I'll update to use your suggested case name.  Me talk pretty one day :)

File: pytest_mark_checker.py (hunk 1 above)
Comment (ryan-rs) [reply pytest_mark_checker.py]: @zreichert The tests you referenced are not py.test, but instead unittest. According to [py.test documentation](https://docs.pytest.org/en/latest/goodpractices.html) the pattern should be `test_` prefixed to test assets.

File: pytest_mark_checker.py (hunk 2 above)
Comment (ryan-rs) [reply pytest_mark_checker.py]: Okay, ticket improvement and I'm cool.

**Pull request merged**